src/edu/stanford/nlp/sempre/SemanticFn.java

```diff
@@ -35,6 +36,12 @@ public void init(LispTree tree) {
     this.tree = tree;
   }
 
+  public SemanticFn copy() {
```

Comment:
SemanticFn is not meant to be copied in general - when is this needed?  Why not just use one object?